### ISSUE TITLE
Add support for original V1 (Vive) lighthouses

### ISSUE THREAD

**Issue by NUKMUK** [jeroen1602/lighthouse_pm#15]

**Comment by jeroen1602** [comment]
I would love to be able to do this, but I don't have any Vive lighthouses to test this with.

So if anyone want to pick this up, be my guest!.

**Comment by jeroen1602** [comment]
I have found a python implementation for what we want, will look into it later to see if I can use this. https://github.com/cinfulsinamon/BasestationPowerManager.

But it will need some work to make the finding of lighthouses more generic because right now it's pretty difficult to change the code to add another supported device.

**Comment by jeroen1602** [comment]
I've got a pr for Vive base stations support, but I can't test if it works.  #35

**Comment by jeroen1602** [comment]
A beta version of the Vive base station support should be merged into master. You can try it by enabling the beta support on the settings page. Feedback on how it works would be appreciated.

**Comment by jeroen1602** [comment]
@BlakCake are you able to make a build or should I make one for you that you can sideload?

**Comment by BlakCake** [comment]
I tried getting along with Flutter but it kinda hates me right now so if you could provide a build, I would appreciate it!

**Comment by jeroen1602** [comment]
You can download a debug build [here](https://www.dropbox.com/s/2ttsmpkrdaw8e1w/lighthouse-debug-build.apk?dl=0).

**Comment by BlakCake** [comment]
Thanks! It doesn't seem to be detecting the base stations. Although the dark mode is looking really good.

![MainScreen](https://i.imgur.com/GQuYjgB.jpg)

Bluetooth setting is showing the base station tho, so my phone _does_ see them

![BluetoothSettings](https://i.imgur.com/4sZ1aLj.jpg)

**Comment by jeroen1602** [comment]
Have you gone into settings to enable Vive Base station support?

**Comment by BlakCake** [comment]
It seems it's working now, thanks for the heads up oops! Sometimes just one base station shows up and I have to Refresh/Restart until they both are showing. I tried putting one in Sleep and got this:

![Oopsie](https://i.imgur.com/5jkSwN7.jpg)

**Comment by jeroen1602** [comment]
It's good to know that the it works, since I don't know what Bluetooth characteristic to read to get the current state. So what you see there is just a guess since I couldn't test what the state should be.

If you want you could try and go through the Bluetooth characteristic's using something like [BLE scanner](https://play.google.com/store/apps/details?id=com.macdom.ble.blescanner) to see what characteristic reports back the current state and what number defines what state.

**Comment by BlakCake** [comment]
You mean this? 

![BLE](https://i.imgur.com/jVaHgQ4.jpg)

**Comment by jeroen1602** [comment]
Yes click connect and go through the services and characteristics to see if one of them changes between on and off (sleep). You only have to go through the characteristics that have the R (read) option. 

P.S. only one device is allowed to connect to a BLE device at a time, so make sure you disconnect the BLE scanner before trying to find the device again.
P.P.S. if you are having scan issues try increasing the scan duration in the settings.

**Comment by BlakCake** [comment]
So I just compared every Custom Service with Read twice between Sleep and On. Nothing seems to be changing between them.

**Comment by jeroen1602** [comment]
Man that sucks, do you know if steam vr does report back the current state?

Also the `0x10ca` (shown in your earlier screenshot) is different from the on command that was being send. So maybe we can use that as the on state.

If there is no way of getting the state then we are stuck to always asking the user what they want to do (which sucks). But this is hopefully mitigated in the future by adding grouping support.

**Comment by NUKMUK** [comment]
Mine seem to change, here's both with SteamVR open
 ![](https://i.imgur.com/Cdi3Uq6.jpg)
![](https://i.imgur.com/OWwGUm5.jpg)
and after closing SteamVR (standby)
![](https://i.imgur.com/VEzSwNr.jpg)
![](https://i.imgur.com/lm4AtNX.jpg)

**Comment by BlakCake** [comment]
I'll try again with SteamVR open

**Comment by jeroen1602** [comment]
So we might get somewhere by reading the `0000CB01-0000-1000-8000-00805F9B34FB` characteristic. and using `0x001503020B` for on and `0x0012000100` for off.

Also steam VR uses a bit of a different method to turn the base stations on. They send an on command with a timeout, and if they don't send the same command again within the timeout then the base station will turn off. But there is also a mode for just turning it on, and the app uses that mode.

**Comment by BlakCake** [comment]
SteamVR open
Unknown State: https://i.imgur.com/K0E15SQ.jpg
After choosing Sleep: https://i.imgur.com/mVPiOx7.jpg
After choosing On: https://i.imgur.com/zBlFpoC.jpg

Everything stopped tracking after this. I may need to restart the base stations

**Comment by jeroen1602** [comment]
@BlakCake I don't see any read properties in your screenshots, did you click the R button?

**Comment by jeroen1602** [comment]
I created a version with a possible fix for the unknown power state characteristic. Download it [here](https://www.dropbox.com/s/whx7bavzay5phmh/lighthouse-debug-build-power-state-fix-29-06-2020.apk). Let me know if it works and if I've missed a state.

**Comment by jeroen1602** [comment]
I would love to know if this build fixes the problem so I can create a new build.

**Comment by Supreeeme** [comment]
Wanted to say it works with powering and turning off my base stations. The state is listed as unknown though.

**Comment by asktoby** [comment]
Just wanted to say I'm using this with my Vive trackers and it is working great! Thank you!

**Comment by RedlineTriad** [comment]
Were these debug features ever merged into the main branch because I don't see a toggle to enable vive base stations and it doesn't seem to find them either.

**Comment by nukmuk** [comment]
I think some recent(?) update broke support for the V1 base stations
